example-app/src/main/java/com/hcaptcha/example/MainActivity.java

```diff
@@ -112,7 +114,14 @@ public void onFailure(HCaptchaException e) {
                         Log.d(TAG, "hCaptcha failed: " + e.getMessage() + "(" + e.getStatusCode() + ")");
                         setErrorTextView(e.getMessage());
                     }
+                })
+                .addOnOpenListener(new OnOpenListener() {
+                    @Override
+                    public void onOpen() {
+                        Toast.makeText(MainActivity.this, "hCaptcha shown", Toast.LENGTH_SHORT).show();
```

Comment:
This is triggered only first time and never when I refresh the challenge. `open-cb` from js api is triggered on challenge refresh but the android listener does not, I believe there is an issue.

Reply:
I faced with problem that `open-callback` don't fire at all. Previously toast showed because of wrong logic.

Also I noted that for Android UI never show pictures to select (to pass captcha), as we have on iOS. on iOS `open-callback` fired without any issues.

@DSergiu @e271828- is there is any relation between `open-callback` calls and the UI representation of captcha? Why for Android we never show 'full captcha' (with pictures to select)? Is it related to testing tokens?

Reply:
It is related to the dev sitekey `10000-...-000001`. You may create your own sitekey via our dashboard https://dashboard.hcaptcha.com and use that sitekey instead.